### PR TITLE
Hoodie plugin

### PR DESCRIPTION
**Pull request by nathanstilwell**
Adding a plugin method as explained in #2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -175,6 +169,15 @@ hoodie.plugin(function (hoodie) {
...
 ```
 
+```js
+  const sayHi = require('sayHi')
+  const grabADrink = require('grabADrink')
+
+  hoodie
+    .plugin(sayHi)
+    .plugin(grabADrink)
+```
```
Comment (gr2m) [inline README.md]: can we add to the example that an object can be passed in, too? Maybe like this?

```js
  const sayHi = require('sayHi')
  const grabADrink = require('grabADrink')

  hoodie
    .plugin(sayHi)
    .plugin({
      grabADrink: grabADrink
    })
```

Comment (nathanstilwell) [inline README.md]: Sure!

File: lib/plugin.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  if (typeof plugin === 'function') {
+    plugin(self)
+  }
+
+  if (typeof plugin === 'object') {
+    for (p in plugin) {
+      if (plugin.hasOwnProperty(p)) {
+        self[p] = plugin[p]
+      }
+    }
```
Comment (gr2m) [inline lib/plugin.js]: Can you do this instead?

```js
Object.keys(plugin).forEach(function (key) {
  self[key] = plugin[key]
})
```

I actually don’t know which one performs better, but this version is simpler to test. Right now, we don’t have a test where `plugin.hasOwnProperty(p)` returns false, and we’d need to extend the Prototype for it which is kinda hacky

Comment (nathanstilwell) [inline lib/plugin.js]: :+1:

**Comment by gr2m** [comment]
Love it, thanks Nathan :bowtie:

**Pull request merged**